src/pocketmine/event/block/SignChangeEvent.php

```diff
@@ -79,7 +79,7 @@ public function getLine($index){
 	 * @param string[] $lines
 	 */
 	public function setLines(array $lines){
-		if(count($lines) > 4){
+		if(count($lines) === 4){
```

Comment:
This condition is inverted. It should be `!== 4`. You're throwing an exception when the line count is correct.